src/Dependencies/Updater.php

```diff
@@ -41,63 +27,74 @@ private function updateDependency($dep)
 
         $type = explode('/', $dep)[0];
         $name = explode('/', $dep)[1];
+
+        $result = $this->prepareForUpdate();
+        if ($result->isErr()) {
+            return $result;
+        }
+
         $depArray = $this->jsonFile->getDependency($type, $name);
         if ($depArray === []) {
             return \Result\Result::err('No matching dependency in whippet.json');
         }
 
-        $this->updateHash();
-        $this->loadGitignore();
+        return $this->update([$type=>[$depArray]]);
```

Comment:
It does look weird at first, but update() expects input in the format:

```
[
    plugins => [
        [
            'name' => 'plugin1',
            'ref' => 'ref1'
        ]
        [
            'name' => 'plugin2',
            'ref' => 'ref2'
        ]
    ]
]
```

So each plugin is its own array, but then they go in an overall array of all plugins. Which means when you have a single plugin, it is a single array within an array. If that makes sense!